### PR TITLE
feat(ip): logContextChanged & logExpandToFullUI events added to the insight client

### PR DESCRIPTION
**Pull request by mmitiche**
## 1. logContextChanged

In the service console inside Salesforce, when a user updates one of the case fields, the insight panel should automatically trigger a new search query and send the ContextChanged search event.

A new `logContextChanged` method is now added to the insight client.

- #### Current ContextChanged event in the C4SF package:

PreviousContextChangedEvent

- #### New ContextChanged event sent using coveo.analytics.js:

MewcontextEvent2

## 2. logExpandToFullUI

In the service console inside Salesforce, when a user opens the full search page, the Hosted Insight Panel should send the ExpandToFullUI  custom event.

A new `logExpandToFullUI` method is now added to the insight client.

- #### Current ExpandToFullUI event in the C4SF package:

CurrentExpandToFullUIEvent

- #### New ExpandToFullUI event sent using coveo.analytics.js:

NewOpenUI2

### CODE REVIEW

**Review by aelfaleh — APPROVED** [review]

**Review by erocheleau — APPROVED** [review]

**Comment by erocheleau** [comment]
Are you merging this PR with this one as well?
https://coveord.atlassian.net/browse/SFINT-4570

**Comment by mmitiche** [comment]
> Are you merging this PR with this one as well?
> https://coveord.atlassian.net/browse/SFINT-4570

Yes I'm planning to do that.

File: src/insight/insightEvents.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export enum InsightEvents {
+    /**
+     * Identifies the search event that gets logged when the query context is updated as a result of updating one of the case context fields.
+     */
+    contextChanged = 'contextChanged',
+    /**
+     * Identifies the event that gets logged when the user opens the full search page from the insight panell.
```
Comment (mmitiche) [inline src/insight/insightEvents.ts:7]:
```suggestion
     * Identifies the event that gets logged when the user opens the full search page from the insight panel.
```

**Comment by mmitiche** [comment]
> It all looks pretty good to me. Using different format for the `customData > caseContext` information might be a problem for reporting though. Including the data of the hosted insight panel into an existing report would be problematic.
> 
> @erocheleau Any thoughts on this?

Totally right! I'm currently working on it to assure that we keeping the exact same format.

**Comment by mmitiche** [comment]
> > It all looks pretty good to me. Using different format for the `customData > caseContext` information might be a problem for reporting though. Including the data of the hosted insight panel into an existing report would be problematic.
> > @erocheleau Any thoughts on this?
> 
> Totally right! I'm currently working on it to assure that we keeping the exact same format.

The events are now updated to exactly match what we currently send in the C4SF package.

File: src/insight/insightClient.spec.ts [hunk 2]
```diff
@@ -253,20 +253,41 @@ describe('InsightClient', () => {
...
 
     it('should send proper payload for #contextChanged', async () => {
-        const meta = {caseId: '1234', caseNumber: '1234', caseContext: {subject: 'test'}};
+        const meta = {
+            caseId: '1234',
+            caseNumber: '5678',
+            caseContext: {subject: 'test subject', description: 'test description'},
+        };
+
+        const metaToBeSent = {
```
Comment (lbergeron) [inline src/insight/insightClient.spec.ts:262]: I'd suggest to name the variable `expectedMeta`. It seems more meaningful to me.

File: src/insight/insightClient.ts [hunk 3]
```diff
@@ -34,6 +34,17 @@ export interface InsightClientOptions extends ClientOptions {
     enableAnalytics: boolean;
 }
 
+const extractContextFromMetadata = (meta: {caseContext: Record<string, string>}) => {
+    const context: Record<string, string> = {};
+    Object.keys(meta.caseContext).forEach((contextKey) => {
+        const capitalizedKey = contextKey.charAt(0).toUpperCase() + contextKey.slice(1);
+        const keyToBeSent = `context_Case_${capitalizedKey}`;
```
Comment (lbergeron) [inline src/insight/insightClient.ts:41]: I'm not sure about adding this `Case_` prefix on all context values. I could imagine an implementation adding some context information that is not directly related to the case. It might look weird to have `Case` in the context key name.

Maybe we could add the `Case_` prefix only for the keys that are well known like `subject` and `description`.

Comment (mmitiche) [reply src/insight/insightClient.ts]: I removed `Case_` prefix. Now we just send `context_${contextKey}`.
If we want to have a `context_Case_Subject` key in the event payload we only have to send a context containing a key named `Case_Subject`

Comment (lbergeron) [reply src/insight/insightClient.ts]: So it'll be the Hosted Insight Panel component responsability to use the same key as before to ensure a consistent reporting.

Comment (mmitiche) [reply src/insight/insightClient.ts]: Yes exactly 👌I will add tests at the Hosted Insight Panel component level to confirm that we always set the correct context keys that will ensure a consistent reporting.
